internal/ingest/httpingest/httpingest.go

```diff
@@ -26,42 +28,84 @@ type Collector interface {
 
 func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	logger := h.getLogger()
+	h.Context = r.Context()
+
+	contentType := r.Header.Get("Content-Type")
+
+	if contentType == "application/cloudevents-batch+json" {
+		err := h.processBatchRequest(w, r)
+		if err != nil {
+			logger.ErrorCtx(h.Context, "unable to process batch request", "error", err)
+			_ = render.Render(w, r, api.ErrInternalServerError(err))
+			return
+		}
+	} else {
+		err := h.processSingleRequest(w, r)
+		if err != nil {
+			logger.ErrorCtx(h.Context, "unable to process single request", "error", err)
+			_ = render.Render(w, r, api.ErrInternalServerError(err))
+			return
+		}
+	}
```

Comment:
I'm not sure about this, @sagikazarmark ?

```suggestion
	var err error
	switch contentType {
	case "application/cloudevents+json":
		err = h.processSingleRequest(w, r)
	case "application/cloudevents-batch+json":
		err = h.processBatchRequest(w, r)
	default:
		_ = render.Render(w, r, api.ErrUnsupportedMediaType(errors.New("content type must be application/cloudevents+json or application/cloudevents-batch+json")))
	}

	if err != nil {
		logger.ErrorCtx(h.Context, "unable to process request", "error", err)
		_ = render.Render(w, r, api.ErrInternalServerError(err))
		return
	}
```